akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/BehaviorSetup.scala

```diff
@@ -121,9 +126,10 @@ private[akka] final class BehaviorSetup[C, E, S](
     } catch {
       case NonFatal(ex) =>
         if (catchAndLog)
-          log.error(ex, s"Error while processing signal [{}]", signal)
+          log.error(s"Error while processing signal [$signal]: $ex", ex)
         else {
-          log.debug(s"Error while processing signal [{}]: {}", signal, ex)
+          if (log.isDebugEnabled)
+            log.debug(s"Error while processing signal [$signal]: $ex", ex)
```

Comment:
Collapse into one if?

Reply:
it's actually a second `throw` on the next line that should alway be done